sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,729 @@
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+struct initial_threshold_key {
+  using value_t = property_value<initial_threshold_key>;
+
+  initial_threshold_key(size_t initialThreshold);
+};
+
+struct maximum_size_key {
+  using value_t = property_value<maximum_size_key>;
+
+  initial_threshold_key(size_t maxSize);
+};
+
+struct read_only_key {
+  using value_t = property_value<read_only_key>;
+
+  read_only_key(bool readOnly);
+};
+
+
+struct zero_init_key {
+  using value_t = property_value<zero_init_key>;
+
+  read_only_key(bool zeroInit);
+};
+
+inline constexpr initial_threshold_key::value_t initial_threshold;
+inline constexpr maximum_size_key::value_t maximum_size;
+inline constexpr read_only_key::value_t read_only;
+inline constexpr zero_init_key::value_t zero_init;
+
+}  // ext::oneapi::experimental
+----
+
+|===
+|Property|Description
+
+|`initial_threshold`
+|The `initial_threshold` property specifies the initial deallocation threshold
+ value for the memory pool. If this property is not used the default value is
+ zero, and this can be increased after the memory pool is created by calling
+ `memory_pool::set_new_threshold`.
+
+|`maximum_size`
+|The `maximum_size` property specifies the maximum size of the memory pool,
+ after which any allocation will result in an exception. If the value specified
+ is larger than the implementation can support an exception with the
+ `errc::memory_allocation` error code is thrown. If this property is not used
+ the default value is implementation-defined.
+
+|`read_only`
+|The `read_only` property is a performance hint which asserts that all memory
+ allocations from the memory pool will only ever be read from, this can be used
+ by the SYCL runtime to optimize for performance.
+
+|`zero_init`
+|The `zero_init` property adds the rquirement that all memory allocated to the
+ memory as it is allocated from the OS to the memory pool will be initialised to
+ zero. Note there is no guarantee that the memory allocation be re-initialized
+ to zero when it is re-allocated from the pool, so users must re-initialize
+ memory to zero if they wish for later allocations to have this behaviour.
+
+|===
+
+
+=== Default memory pools
+
+As well as being able to construct a memory pool explicitly, this extension
+introduces a default memory pool per device for each SYCL context and device
+pair for device allocations and a default memory pool per context for host
+allocations.
+
+New member functions are added to the `context` class to retrieve the default
+memory pool as a copy of the `memory_pool` object. This can be modified and have
+those modifications reflected as it conforms to the SYCL common reference
+semantics.
+
+[source,c++]
+----
+class context {
+
+  memory_pool context::ext_oneapi_get_default_memory_pool();
+
+  memory_pool context::ext_oneapi_get_default_memory_pool(device dev);
+
+}; // context
+----
+
+[source, c++]
+----
+memory_pool context::ext_oneapi_get_default_memory_pool();
```

Comment:
```suggestion
memory_pool context::ext_oneapi_get_default_memory_pool() const; 
```
Not sure if these can be `const` if they are query methods

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/2c51adf28ece21c0e91c2c0ec85440ae1153e536.